src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -360,9 +360,36 @@ private static string[] HandleClear(string[] input)
             return input;
         }
 
-        // Trying to fetch a property value from tfm property bags.
-        // If defined the property should have identical values in all of the occurances.
-        private static TValue GetNonEvaluatedPropertyOrNull<TValue>(
+        private static TValue GetSingleOrDefaultProperties<TValue>(
+            IVsTargetFrameworks tfms,
+            string propertyName,
+            Func<string, TValue> valueFactory)
+        {
+            var warningProperties = GetNonEvaluatedPropertyOrNull(tfms, propertyName, valueFactory);
+
+            if (warningProperties.Count() > 1)
```

Comment:
style: simplified it to check if count is 1 then return `warningProperties.First()` else return `default(value)`

Reply:
style: simpler would be - 

`return warningProperties.Count() == 1 ? warningProperties.First : default(TValue)`